homeassistant/components/whois/sensor.py

```diff
@@ -121,6 +145,13 @@ def _ensure_timezone(timestamp: datetime | None) -> datetime | None:
         entity_registry_enabled_default=False,
         value_fn=lambda domain: getattr(domain, "reseller", None),
     ),
+    WhoisSensorEntityDescription(
+        key="status",
+        translation_key="status",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value_fn=lambda domain: _get_status_type(domain.status),
```

Comment:
Shouldn't it just be:

```suggestion
        value_fn=lambda domain: STATUS_TYPES.get(domain.status),
```

Additionally, we need to set the list of possible options for this sensor and set the device class to enum to indicate this is an enumeration.

Reply:
About the enum; i agree. Updated.

About the `STATUS_TYPES.get`, that won't work, since the whois server can return different strings. E.g. some servers appends the icann url to the result, others don't. So I now look if the icann defined string is present in the returned value (in the function `_get_status_type`).